x/auth/client/query.go

```diff
@@ -134,26 +135,22 @@ func getBlocksForTxResults(clientCtx client.Context, resTxs []*ctypes.ResultTx)
 }
 
 func formatTxResult(txConfig client.TxConfig, resTx *ctypes.ResultTx, resBlock *ctypes.ResultBlock) (*sdk.TxResponse, error) {
-	anyTx, err := parseTx(txConfig, resTx.Tx)
+	anyTx, err := txToAny(txConfig, resTx.Tx)
 	if err != nil {
 		return nil, err
 	}
 
-	return sdk.NewResponseResultTx(resTx, anyTx.AsAny(), resBlock.Block.Time.Format(time.RFC3339)), nil
+	return sdk.NewResponseResultTx(resTx, anyTx, resBlock.Block.Time.Format(time.RFC3339)), nil
 }
 
-func parseTx(txConfig client.TxConfig, txBytes []byte) (codectypes.IntoAny, error) {
-	var tx sdk.Tx
-
-	tx, err := txConfig.TxDecoder()(txBytes)
+func txToAny(txConfig client.TxConfig, txBytes []byte) (*codectypes.Any, error) {
+	txb, err := txConfig.TxDecoder()(txBytes)
 	if err != nil {
 		return nil, err
 	}
-
-	anyTx, ok := tx.(codectypes.IntoAny)
+	p, ok := txb.(tx.ProtoTxProvider)
```

Comment:
I think the edge case is when using a TxConfig that's a StdTxConfig, this will fail because StdTx doesn't implement ProtoTxProvider.

iirc, I believe that's why I initially chose the IntoAny hack, because `Any` is something that both wrapper and StdTx understand.

In practice, I think having an amino-only chain doesn't work anyways, so maybe it's fine to drop support for StdTxConfig in this case?

Reply:
Thanks for note. So what we need is a `Tx` (interface) provider. I will have a look into it tomorrow.